src/test/scala/com/advancedtelematic/director/http/AssignmentsResourceSpec.scala

```diff
@@ -173,6 +212,21 @@ class AssignmentsResourceSpec extends DirectorSpec
     msg shouldBe defined
   }
 
-  // TODO: Cancel Implementation missing
```

Comment:
We need to publish and test `DeviceUpdateCanceled`, I think campaigner picks it up

Reply:
It's published in DeviceAssignments.scala and the parent class is tested in the test "PATCH assignments cancels assigned updates". I can add a test for the correct subclass.